tests/version_test.py

```diff
@@ -0,0 +1,69 @@
+"""Check that chart versions have been updated."""
+
+from __future__ import annotations
+
+from io import BytesIO
+from pathlib import Path
+
+import requests
+import yaml
+from git import DiffIndex, Repo
+from semver import VersionInfo
+
+REPOSITORY_URL = "https://lsst-sqre.github.io/charts/index.yaml"
+"""The URL for the Helm repository index of currently-published charts."""
+
+
+def test_versions(tmp_path: Path) -> None:
+    """Check that all modified charts have an updated version number.
+
+    For automatic publication of new Helm charts to work, we have to remember
+    to update the version number of a chart whenever we make changes to it.
+    This test verifies that by verifying there have been no Git commits that
+    change the contents of a chart since the last tagged release if the
+    version of the chart matches the last released version.
+
+    This check is done with Git rather than by comparing the contents of the
+    package since helm package makes changes to files when packaging (such as
+    canonicalizing Chart.yaml).
+    """
+    root_path = Path(__file__).parent.parent
+    charts_path = root_path / "charts"
+    repo = Repo(str(root_path))
+
+    r = requests.get(REPOSITORY_URL)
+    assert r.status_code == requests.codes.ok
+    index = yaml.safe_load(r.text)
+
+    # Only check charts that have been published.
+    published_charts = set(index["entries"].keys())
+    current_charts = set((p.name for p in charts_path.iterdir()))
+    charts_to_check = published_charts & current_charts
+
+    errors = []
+    for chart in charts_to_check:
+        published = max(
+            index["entries"][chart],
+            key=lambda e: VersionInfo.parse(e["version"])
+        )
+
+        # If the version number in the current repository is later than the
+        # latest published version, differences are allowed.
+        with (charts_path / chart / "Chart.yaml").open() as f:
+            metadata = yaml.safe_load(f)
+            current = VersionInfo.parse(metadata["version"])
+            latest = VersionInfo.parse(published["version"])
+            if current > latest:
+                continue
+
+        # Otherwise, construct the tag of the previous release and look for
+        # any changes in Git since that release.
+        tag = f"{chart}-{published['version']}"
+        diff = repo.head.commit.diff(tag, paths=[f"charts/{chart}"])
```

Comment:
if this tag doesn't exist just it just return None?  Or for a new chart do we push the first tag?  I'm guessing this just keeps going if the tag doesn't exist

Reply:
It shouldn't be possible for the tag to not exist because if we reached this point the prior release appears in the repository index, which is driven by the tags. But I'll rewrite this to not fail if the tag doesn't exist, just in case. Good catch.